_posts/blog/2016-09-13-code-reviews.md

```diff
@@ -0,0 +1,91 @@
+---
+layout: post
+title: "Handbook Of Manners: How To Behave In A Code Review"
+description: Some thoughts on code reviews and the way they usually are and they should be performed.
+modified:
+categories: blog
+author: henning_gross
+excerpt: Code reviews are a common practise for knowledge distribution, learning and increasing quality - but do we perform them in a way that actually creates value?
+tags: [Mindset, Culture, Agile, XP]
+date: 2016-09-13T17:49:00-04:00
+---
+
+In most companies __code reviews__ are part of the __process__ today. In theory they help distributing knowledge, onboarding new people, increasing learning and quality.
+
+
+A code review, no matter if it is performed in a pairing manner or asynchronously is a situation in which one person __inspects__ the work of another person.
+
+
+This can lead to the __assumption__ that the reviewer has to accept the changes the reviewee wants to introduce and put the reviewer in a position of power over the reviewee who has to __seek for approval__.
+
+
+__Agile development teams__ strive for a conception and identity in which everyone is __equally__ important, equally __worthy__ and hierarchies usually make __no difference__ on functional matters (as long as the manager does not have to protect the business but that's another story).
+
+
+So how does a code-review ideally look like?
+
+
+Let's base on the assumption that every reviewer has the same intent - which is to __convince__ the reviewee to follow the suggested changes (to improve the quality of the added code).
+
+
+When this is the main __intent__, different approaches to code reviews can only be different __strategies__ to achieve the same goal.
+
+
+The following strategies/behavioural advices can help to achieve this goal and at the same time improve the team spirit and culture.
+
+
+## Consult - Don't Preach
+
+The basic mindset in which an engineer should approach a code review for a change introduced by another engineer is the mindset of a __consultant__. Whatever you say or write: do it with caution. Sentences that start with _I would approach the problem this way..._ or _I believe this way would be more efficient_ are a good approach if you want to encourage the reviewee to accept your recommendations. Way better than _This solution is inappropriate!_ or _Do it like this!_. Command-style comments are guaranteed to result in damage, conflicts, decreased creativity and motivation and bad team spirit. Don't do it.
```

Comment:
Really?! Why semi-colon? That looks really strange to me. I of course follow but can you please explain?

Also please explain the irony. It might be unintended...